libs/ts-rest/nest/src/lib/ts-rest-nest.ts

```diff
@@ -1,33 +1,49 @@
-import { AppRoute, AppRouter, ApiRouteResponse, Without, getRouteResponses } from '@ts-rest/core';
-import { ApiDecoratorShape } from './api.decorator';
+import {
+  AppRoute,
+  AppRouter,
+  ApiRouteResponse,
+  Without,
+  getRouteResponses,
+  ApiResponseForRoute,
+} from '@ts-rest/core';
+import { TypedRequestShape } from './typed-request.decorator';
 
 type AppRouterMethodShape<T extends AppRoute> = (
   ...args: any[]
 ) => Promise<ApiRouteResponse<T['responses']>>;
 
-type AppRouterControllerShape<T extends AppRouter> = {
-  [K in keyof T]: T[K] extends AppRouter
-    ? undefined
-    : T[K] extends AppRoute
-    ? AppRouterMethodShape<T[K]>
-    : never;
-};
+type AppRouterControllerShape<T extends AppRouter> = Without<
+  {
+    [K in keyof T]: T[K] extends AppRoute ? AppRouterMethodShape<T[K]> : never;
+  },
+  never
+>;
 
-type AppRouteShape<T extends AppRouter> = {
-  [K in keyof T]: T[K] extends AppRouter
-    ? AppRouteShape<T[K]>
-    : T[K] extends AppRoute
-    ? ApiDecoratorShape<T[K]>
-    : never;
-};
+type AppRouterRequestShapes<T extends AppRouter> = Without<
+  {
+    [K in keyof T]: T[K] extends AppRoute ? TypedRequestShape<T[K]> : never;
+  },
+  never
+>;
+
+type AppRouterResponseShapes<T extends AppRouter> = Without<
+  {
+    [K in keyof T]: T[K] extends AppRoute ? ApiResponseForRoute<T[K]> : never;
+  },
+  never
+>;
 
-export type NestControllerShapeFromAppRouter<T extends AppRouter> = Without<
+type NestControllerShapeFromAppRouter<T extends AppRouter> = Without<
   AppRouterControllerShape<T>,
   AppRouter
 >;
 
-export type NestAppRouteShape<T extends AppRouter> = AppRouteShape<T>;
```

Comment:
We should communicate we are removing this from exports. If people are using this (somehow), that's a breaking change.

Reply:
I believe this was an unintended export, as it was never meant to be used directly. I'm going back to @oliverbutler's comment on avoiding breaking changes the second we document our API (which we did but not too long ago, I doubt anyone picked this one up specifically from the docs yet 😅).

I'm personally fine with the trade-off of potentially breaking some code using undocumented features for the purpose of keeping our code more maintainable. Especially since this specific example would be very easy to fix.

I guess, rather than bumping the major version for such a small change, we could at least add it to the release notes, in case this affects some people.

Reply:
Yup, that was my point, let's at a minimum, add it to the release notes.